File: py_opencv/blur.py
import cv2 as cv
import numpy as np

img = cv.imread("photos/dog1.jpg")
img = cv.resize(img, (int(img.shape[1] * 0.2), int(img.shape[0]*0.2)), interpolation=cv.INTER_LINEAR)
cv.imshow("dog",img)

# Averaging 
average = cv.blur(img, (3,3))
cv.imshow("averaged ", average)

# Gaussian
gaussian = cv.GaussianBlur(img, (3,3), 0)
cv.imshow("gaussian", gaussian)

# median blur
median = cv.medianBlur(img, 3)
cv.imshow("median", median)

# bilateral blurring (most effective) blurred, but also preserve edges
bilateral = cv.bilateralFilter(img, 5, 100, 100)
cv.imshow("bilateral", bilateral)

cv.waitKey(0)

File: cpp_opencv/blur/blur.cpp
#include <opencv2/opencv.hpp>
#include <vector>
#include <stdio.h>
#include <string>
#include <vector>

using namespace cv;

std::string photo = "../../photos/dog1.jpg";

int main(){
    Mat image = imread(photo);
    resize(image, image, cv::Size(), 0.2, 0.2);
    imshow("dog", image);

    // Averaging 
    Mat avg;
    blur(image, avg, Size(3, 3));
    imshow("average", avg);

    // Gaussian
    Mat gau; 
    GaussianBlur(image, gau, Size(3,3), 0);
    imshow("gaussian", gau);

    // Median
    Mat med;
    medianBlur(image, med, 3);
    imshow("median", med);

    // Bilateral 
    Mat bi;
    bilateralFilter(image, bi, 5, 100, 100);
    imshow("bilateral", bi); 


    waitKey(0);
    return 0;
}

